pkg/nmi/server/endpointpolicy.go

```diff
@@ -0,0 +1,175 @@
+// +build windows
+
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	v1 "github.com/Microsoft/hcsshim"
+	msg "github.com/Microsoft/hcnproxy/pkg/types"
+	client "github.com/Microsoft/hcnproxy/pkg/client"
+	"k8s.io/klog"
+)
+
+// ApplyEndpointRoutePolicy applies the route policy against the pod ip endpoint
+func ApplyEndpointRoutePolicy(podIP string, metadataIP string, metadataPort string, nmiIP string, nmiPort string) error {
+	if len(podIP) <= 0 {
+		return errors.New("Missing IP Address")
+	}
+
+	endpoint, err := getEndpointByIP(podIP)
+	if err != nil {
+		return err
+	}
+
+	er := addEndpointPolicy(endpoint, metadataIP, metadataPort, nmiIP, nmiPort)
+	if er != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteEndpointRoutePolicy applies the route policy against the pod ip endpoint
+func DeleteEndpointRoutePolicy(podIP string, metadataIP string) error {
+	if len(podIP) <= 0 {
```

Comment:
><= [](start = 15, length = 2)

I see you use it in more than one place - string cannot have negative length

Reply:
Update it to be podIP == "". Let me know if you have any conceens. @ 

---
In reply to: [383554501](https://github.com/bowu24/aad-pod-identity/pull/1#discussion_r383554501) [](ancestors = 383554501)